### DISCUSSION TITLE
How to avoid Error: EACCES: permission  with wsl.exe spammy messages:

### DISCUSSION THREAD

**Discussion by ssbarnea** [Vampire/setup-wsl#67] [Q&A] [answered]
Something recent is causing lots of messages like below on each shell command:
```
Unexpected error attempting to determine if executable file exists 'C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps\wsl.EXE': Error: EACCES: permission denied, stat 'C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps\wsl.EXE'
```

Apparently the execution still happens but it makes much harder to read the console logs. Example https://github.com/ssbarnea/mise-action/actions/runs/12549125315/job/34989613798?pr=1

I tried to google about this error in particular but I was not able to find anything useful yet.

**Comment by Vampire** [comment]
For using WSLv2 on `windows-2022` or any WSL version on `windows-2025` currently for the former the action needs to do a `wsl --update` for the latter a `wsl --install`.

Both put the new link `C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps\wsl.exe` into place which happens to be on the `PATH`, but somehow the file is not properly accessible.

On a normal computer I'd say reboot.

Here again I hope https://github.com/actions/runner-images/issues/10563 and https://github.com/actions/runner-images/issues/11265 get done so that the file is already there and accessible which would also fix it.

In the meantime I added some logic to delete the file after the install / update, as it is anyway not the one used from `PATH`.
This will be included in the next version of the action.

As a work-around currently, you can switch to use WSLv1, or delete that link it complains about.

**Reply by ssbarnea** [reply]
Funny is that while digging, I found https://github.com/actions/toolkit/issues/1925 raised by you, better to continue there.

But you could also do a `rm C:\Users\runner\AppData\Local\Microsoft\WindowsApps\wsl.exe" after wsl is installed to remove the symlink, wsl.exe will still be in path.

**Reply by Vampire** [reply]
I raised that right before answering here just in case it is a bug on their side which I'm not sure of.
And indeed if the files were not there before doing the `wsl --install` I delete them after the `wsl --install` finished as work-around.
I wrote that above in the next to last paragraph. :-)

**Reply by Vampire** [reply, marked as answer]
Thank you very much for that pull request, it indeed fixes the problem properly.
I monkey-patched your PR's changes into the build result of this action and was able to remove the file-deleting.
Now theses annoying and unnecessary "errors" are gone.
I just released a new version of the action that includes the fix.
